src/lib.rs

```diff
@@ -266,6 +273,10 @@ pub enum MiriMemoryKind {
     Env,
     /// mutable statics
     MutStatic,
+    /// in-memory file descriptors
+    MemFd,
+    /// The stack allocation
```

Comment:
It took me a bit to understand what you are doing. This needs more comments.

The goal is to have a fake allocation that is returned by the "give me the stack root" call, but otherwise entirely unused, right?